network/gossip/libp2p/libp2pNode_test.go

```diff
@@ -342,9 +340,86 @@ func (l *LibP2PNodeTestSuite) TestCreateStreamTimeoutWithUnresponsiveNode() {
 	assert.Error(l.T(), err)
 }
 
+// TestCreateStreamIsConcurrent tests that CreateStream calls can be made concurrently such that one blocked call
+// does not block another concurrent call.
+func (l *LibP2PNodeTestSuite) TestCreateStreamIsConcurrent() {
+
+	// bump up the unicast timeout to a high value
+	unicastTimeout = time.Hour
+
+	// create two regular node
+	goodPeers, goodAddrs := l.CreateNodes(2, nil, false)
+	defer l.StopNodes(goodPeers)
+	require.Len(l.T(), goodAddrs, 2)
+
+	// create a silent node which never replies
+	listener, silentNodeAddress := newSilentNode(l.T())
+	defer listener.Close()
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	ch := make(chan struct{})
+	// spin off a go routine to create a stream to the unresponsive node
+	go func() {
+		wg.Done()
+		_, _ = goodPeers[0].CreateStream(l.ctx, silentNodeAddress) // this call will block
+		close(ch)
+	}()
+
+	// make sure the go routine to create a stream with the unresponsive node actually started
+	unittest.AssertReturnsBefore(l.T(), wg.Wait, 5*time.Millisecond)
+	// make sure the go routine to create a stream did not finish
+	select {
+	case <-time.After(10 * time.Millisecond):
+	case <-ch:
+		assert.Fail(l.T(), "CreateStream attempt to the unresponsive peer did not block")
+	}
+
+	var err error
+	// assert that the same peer can still connect to the other regular peer without being blocked
+	unittest.AssertReturnsBefore(l.T(),
+		func() {
+			_, err = goodPeers[0].CreateStream(l.ctx, goodAddrs[1])
+		},
+		10*time.Millisecond)
+	assert.NoError(l.T(), err)
```

Comment:
We could also check after here to ensure that the other call is still blocked:
```go
select {
    case <-ch:
        // fail
    default:
}
```

Reply:
that is an excellent suggestion.